docs/enhancements/tenancy-based-auth.md

```diff
@@ -0,0 +1,83 @@
+
+## Goals
+The following tables details the access permission rules for various resources:
+
+For clusters and cluster-based resources (all APIs that starts with `/v2/clusters/{cluster_id}`):
+| Action         | Authorization Access                                                                           |
+|----------------|------------------------------------------------------------------------------------------------|
+| List Cluster   | Returns clusters owned by all members of the organization                                      |
+| Get  Cluster   | Allowed for all members of the organization                                                    |
+| Edit Cluster   | Allowed for the owner and for each user that has update access for this cluster's subscription |
+| Delete Cluster | Allowed for the owner and for each user that has remove access for this cluster's subscription |
+
+For InfraEnv and InfraEnv-based resources like hosts (all APIs that starts with `/v2/infra-envs/{infra_env_id`):
+| Action         | Authorization Access                                                                           |
+|----------------|------------------------------------------------------------------------------------------------|
+| List InfraEnvs | For bound InfraEnvs, returns resources owned by all members of the organization<br>If unbound InfraEnvs, return resources if owned by the caller                                                                |
+| Get    InfraEnv| For bound InfraEnv, allowed for all members of the organization<br>For unbound InfraEnv allowed for the owner                                                                                                         |
+| Edit InfraEnv  | For bound InfraEnv, allowed for the owner and for each user that has update access<br>for the bound cluster's subscription. For unbound InfraEnv, allowed for the owner                                               |
+| Delete InfraEnv| For bound InfraEnv, allowed for the owner and for each user that has remove access<br>for the bound cluster's subscription. For unbound InfraEnv, allowed for the owner                                               |
+
+For day2 cluster operations:
+| Action         | Authorization Access                                                                           |
+|----------------|------------------------------------------------------------------------------------------------|
+| Add Host       | Allowed for every user who has update access for the day1's cluster subscription               |
+
+Filtering for you own clusters:
+Get Cluster call shall have an additional query parameter which filters the cluster by owners only:
+`GET /v2/clusters?owneronly=true`
+
+## Authorization Flow
+![authorization flow](./auth_flow.png)
+
+
+Upon receiving a request from a client, the authorization layer (acting as a reverse proxy) issues an access_review and capability_review requests to ACM to establish the user/agent role:
```

Comment:
Again, AMS.

We have the authorization layer already, right?

Reply:
Yes. It will required some small alterations in the logic but the flow remains the same